### PR TITLE
[rust] update rules_rust and remove workaround

### PR DESCRIPTION
**Pull request by nbdd0121**
This bumps rules_rust to a version rebased on 0.30, with fixes so that renamed disabled optional dependency of `serde_with` won't cause trouble.

### CODE REVIEW

File: third_party/rust/repos.bzl [hunk 1]
```diff
@@ -95,9 +95,9 @@ def rust_repos(rules_rust = None, safe_ftdi = None, serde_annotate = None):
     http_archive_or_local(
         name = "rules_rust",
         local = rules_rust,
-        sha256 = "e85891dd6bb24e877a9622c09f028b9e11c0f4df69db4735194eb1d8dc7ae917",
-        strip_prefix = "rules_rust-rebase-20230822_02",
-        url = "https://github.com/lowRISC/rules_rust/archive/refs/tags/rebase-20230822_02.tar.gz",
+        sha256 = "ef99111160be57c1f626698f62bc522d17e95ee36af3a026a26da88777650723",
+        strip_prefix = "rules_rust-6c3ffa8c4d8e07c94c572e816797f53b61fabaec",
+        url = "https://github.com/lowRISC/rules_rust/archive/6c3ffa8c4d8e07c94c572e816797f53b61fabaec.tar.gz",
```
Comment (nbdd0121) [inline third_party/rust/repos.bzl:100]: Waiting for a new branch and a new tag to be created

Comment (timothytrippel) [reply third_party/rust/repos.bzl]: Just to confirm, your new branch / tag will inclue the same patches already present on: https://github.com/lowRISC/rules_rust/commits/rebase-20230822 ?

(Just checking b/c these patches are workarounds for the nightly regression infra and EDA tool infra at other partner sites too.)

Comment (nbdd0121) [reply third_party/rust/repos.bzl]: Here's the list of commits contained: https://github.com/nbdd0121/rules_rust/commits/main

I think there are only 3 patches (excluding my new fixes) on top?

Comment (timothytrippel) [reply third_party/rust/repos.bzl]: awesome, yes that looks good. thanks.

**Review by timothytrippel — APPROVED** [review]
LGTM as well, as soon as the new branch / tag is created.

**Comment by nbdd0121** [comment]
Switched to tag and ready to merge

**Pull request merged**